include/irr/asset/ICPUMeshBuffer.h

```diff
@@ -149,6 +146,31 @@ class ICPUMeshBuffer : public IMeshBuffer<ICPUBuffer, ICPUDescriptorSet, ICPURen
             return false;
         if (normalAttrId != other->normalAttrId)
             return false;
+        if (m_indexBufferBinding.offset != other->m_indexBufferBinding.offset)
+            return false;
+        if ((!m_indexBufferBinding.buffer || !other->m_indexBufferBinding.buffer) && m_indexBufferBinding.buffer != other->m_indexBufferBinding.buffer)
+            return false;
+        if (!m_indexBufferBinding.buffer->canBeRestoredFrom_recurseDAG(other->m_indexBufferBinding.buffer.get()))
+            return false;
+        for (uint32_t i = 0u; i < MAX_ATTR_BUF_BINDING_COUNT; ++i)
+        {
+            if (m_vertexBufferBindings[i].offset != other->m_vertexBufferBindings[i].offset)
+                return false;
+            if ((!m_vertexBufferBindings[i].buffer || !other->m_vertexBufferBindings[i].buffer) && m_vertexBufferBindings[i].buffer != other->m_vertexBufferBindings[i].buffer)
+                return false;
+            if (!m_vertexBufferBindings[i].buffer->canBeRestoredFrom_recurseDAG(other->m_vertexBufferBindings[i].buffer.get()))
+                return false;
+        }
+
+        if ((!m_descriptorSet || !other->m_descriptorSet) && m_descriptorSet != other->m_descriptorSet)
+            return false;
+        if (!m_descriptorSet->canBeRestoredFrom_recurseDAG(other->m_descriptorSet.get()))
+            return false;
+
+        if ((!m_pipeline || !other->m_pipeline) && m_pipeline != other->m_pipeline)
+            return false;
+        if (!m_pipeline->canBeRestoredFrom_recurseDAG(other->m_pipeline.get()))
```

Comment:
pipeline is not optional IIRC

Reply:
it is, it's not used at all right now in a way other than returning pipeline from mesh loader

Reply:
you cant draw a meshbuffer without a pipeline, you dont even know what primitive type it is

Reply:
i mean `video` namespace ignores it

Reply:
!?

AFAIK its an error to have a ICPUMeshBuffer without a valid ICPUPipeline

Reply:
never heard of this really